### PR TITLE
Planemo integration: Testing tools from VSCode

### PR DESCRIPTION
**Pull request by davelopez**
Ref #95 

## Planemo basic integration

This sets the foundation to interact with planemo from the VSCode Galaxy Tools extension.

A new `planemo` icon has been added to the VSCode activity bar (on the left). For the moment it just displays a welcome message and provides a shortcut button to go directly to the settings related to the planemo configuration.

image

In the new settings, you can point to the planemo binary that will be used to launch all the planemo commands and you can also set the galaxy root directory used by some of them.

## Planemo Tests

The first feature included in this PR is the ability to run tests for the currently opened tool directly from the `Test Explorer`. 
- The tests are automatically discovered by the [galaxy-language-server](https://github.com/galaxyproject/galaxy-language-server/compare/planemo_integration?expand=1#diff-19c8682cdaa16749a015f5b181b2873347c06772c5afbdf2e1498e0c4e70f638R197) when you open a tool or save the tool document (this can be changed in the settings).
- You can then run all the tests from the `Test Explorer` by using `planemo test` in the background. Currently running individual tests is not supported as AFAIK `planemo` does not have an option to do so at the moment.
- After successfully running the tests, the results will be displayed in a convenient way directly on your source XML. 

The failing tests will be marked in red and the reason for failure can be seen directly beside the test definition in the same line or more detailed in the `Output`. You can also directly navigate to each of the tests XML source from the `Test Explorer`.
This can be very convenient especially when having a large number of tests in your tool.

PlanemoTest02

The test integration in VSCode has been possible by implementing a [Test Adapter](https://github.com/hbenl/vscode-test-adapter-api) for the [Test Explorer UI Extension](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-test-explorer).

## New requirements

The [Test Explorer UI Extension](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-test-explorer) will be now a **required dependency extension** that needs to be installed alongside the Galaxy Tools extension if the user doesn't have it already installed.

### CODE REVIEW

**Comment by jmchilton** [comment]
Amazing!

File: client/README.md [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
-In order to use the [Galaxy Language Server](https://pypi.org/project/galaxy-language-server/) features you need Python 3.8+ installed on your system. See the [Installation](#Installation) section for more details.
+# Requirements
+
+## Requires ![PyPI - Python Version](https://img.shields.io/pypi/pyversions/galaxy-language-server)
+
+To use the [Galaxy Language Server](https://pypi.org/project/galaxy-language-server/) features you need Python 3.8+ installed on your system. See the [Installation](#Installation) section for more details.
+
+## Planemo
+
+Since version `0.4.0` you can use some of the cool features of [planemo](https://github.com/galaxyproject/planemo) directly from the extension. You only need to provide the required path to your planemo installation (see [configuration](#planemo-settings)) and the necessary parameters.
```
Comment (bgruening) [inline client/README.md:15]: can the extension by default just assume that planemo in on PATH?

Comment (davelopez) [reply client/README.md:15]: Good idea! done! ;)

**Review by bgruening — APPROVED** [review]
Ok, let's shit this. This will enable so many additional things in the long run now ... amazing! 

Thanks a lot!

**Pull request merged**